pydantic/generics.py

```diff
@@ -131,6 +135,7 @@ def __class_getitem__(cls: Type[GenericModelT], params: Union[Type[Any], Tuple[T
         _generic_types_cache[(cls, params)] = created_model
         if len(params) == 1:
             _generic_types_cache[(cls, params[0])] = created_model
+            _limit_cache_size(_generic_types_cache, _generic_types_cache_limit)
```

Comment:
possibly an alternative solution that doesn't require maintainers to remember to call this cache limiter every time the dict is modified, whilst recalling the reason for avoiding lru_cache - 

could the cache limiting be implemented directly on the dict? presumably as a subclass of dict (or userdict) so that anyone relying on introspection of the dict would see the same api

i.o.w override `__setitem__` ?

Reply:
good idea, though I wonder how many people are really using `_generic_types_cache`, through probably still worth doing for the same of correctness.

Reply:
Done, see c04cd8e.

Reply:
Well that was much slower and uglier than I had hoped, cython complained, but only on CI (I couldn't replicate the exception locally on Ubuntu) so I had to stop inheriting from `dict` or `UserDict`.

Reply:
:disappointed: this is very annoying, last try.